### ISSUE TITLE
Persisted Grants and Invalid Grant Error

### ISSUE THREAD

**Issue by billcunnien** [DuendeSoftware/Support#783]
Good morning!

IS v6.2.3
.NET 6.0

I have added the in-memory persisted grant store (.AddInMemoryPersistedGrants()) and when I attempt to use the refresh token to get a new access token, I am getting the invalid grant error.

I am not quite sure how to resolve this. The log entry is this:

2023-07-20 11:31:43.111 -04:00 [DBG] authorization_code grant with value: ***** not found in store.

Is there something else that I need to do with the grant store setup?

Thanks!
Bill

**Comment by billcunnien** [comment]
At first, I thought I needed to initiate the persisted grant store (in memory or other), but what I discovered is that the connect/token endpoint does have different key/value pairs for a refresh token request that I overlooked. Once I fixed those, then the refresh token request was successful. That leads me to a follow-up question - once this moves into a load balanced situation, will we need to implement that persisted grant store? Would a persisted grant store work with Redis or something similar?
